### ISSUE TITLE
UnknownNextPeer when some nodes are offline, even if other routes exist

### ISSUE THREAD

**Issue by kulpreet** [lightningnetwork/lnd#1304]
### Background

Three lnd nodes, A, B and C. Created channels between A - B and B - C. With both local and remote balances in both the channels. There were also some other channels created by autopilot on A and C (note, not on B).

On trying to send single payments from A to B and then from B to C, things worked fine.

When I ran queryroute from A with dest set to C and some amount that was under the balances, I got a few routes. None of the routes include just using the hops from A-B-C, i.e. all the hops are through channels set up by autopilot.

However, on trying to make a payment I got the UnknownNextPeer error.

Later, I closed all channels that were opened by autopilot and tried the payment again and it went through.

Also, now, query routes to C run from A shows direct route through B, i.e. A-B-C is now the one and only route returned by queryroutes.

*Question:* When running sendpayments, maybe user should be able to provide a "hint" and/or "constraints" about using routes that include a specific node?

Other notes:

* bitcoind was running with txindex on all three nodes, referring to https://github.com/lightningnetwork/lnd/issues/568

* Some of the channels on C opened by autopilot had to be force closed because I got the peer is offline error.

`[lncli] rpc error: code = Unknown desc = unable to gracefully close channel while peer is offline (try force closing it instead): channel link not found`

### Your environment

* version of `lnd` 
`lnd version 0.4.1-beta commit=321cc69e4e93ae5a176e6649702362bd2b7e38e8`
* which operating system (`uname -a` on *Nix)
`Linux ip-172-30-4-30 4.4.0-1060-aws #69-Ubuntu SMP Sun May 20 13:42:07 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
* version of `btcd`, `bitcoind`, or other backend
`Bitcoin Core Daemon version v0.16.0`
* any other relevant environment details
`txindex enabled, if that helps`

### Steps to reproduce

Tell us how to reproduce this issue. Please provide stacktraces and links to code in question.

- setup three nodes with autopilot enabled using bitcoind
- fund the three nodes to open more channels
- open channels using --local_amt and --push_amt from A to B and then from B to C
- try making payment from A to C, the payment fails as there is no route from A-B-C

### Expected behaviour

Multi-hop payment should have successfully gone through even with some nodes were offline. 

In other words, if channels opened by autopilot go offline, that should not impact the use of routes setup using channels explicitly opened by users.

### Actual behaviour

If some nodes go offline for routes between nodes, even if there are other routes available, multi-hop payment fails.

**Comment by Roasbeef** [comment]
A release was very recently made. Are you able to reproduce running `0.4.2`?

> None of the routes include just using the hops from A-B-C, i.e. all the hops are through channels set up by autopilot.

`queryroutes` just looks at static graph information. If the channels hadn't yet been fully advertised (6+ confs), then that information wouldn't be available to it. 

> However, on trying to make a payment I got the UnknownNextPeer error.

Atm, we only return the final error encountered, if you look in the log, there may have been other error encounter. 

> Also, now, query routes to C run from A shows direct route through B, i.e. A-B-C is now the one and only route returned by queryroutes.

If this is after all the channels have been fully advertised, then that makes sense. 

> bitcoind was running with txindex on all three nodes, referring to #568

This may be part of the issue. It isn't safe (atm) to run `bitcoind` instance that supports _multiple_ `lnd` nodes unless you also configure the number of `rpcthreads` on the `bitcoind` node. The documentation briefly cover this.

**Comment by kulpreet** [comment]
I'll try 4.2 later today. But just to be sure I could replicate the problem in 4.1.

I have three nodes on three different machines, talking to separate bitcoind instances.

I close all channels on all three nodes.

Then ran node A and C with autopilot, and ran B without autopilot.

I query for a route from A to C and it get the routes in the file attached.

Then I addinvoice on C and finally sendpayment from A and I get a fair few errors. Also attaching the relevant parts from my lnd debug log on A.

I know this all might be completely useless once i upgrade to 4.2, but wanted to capture it somewhere before I upgrade.


[routes-a-to-c.json.tar.gz](https://github.com/lightningnetwork/lnd/files/2077051/routes-a-to-c.json.tar.gz)
[lng.log.tar.gz](https://github.com/lightningnetwork/lnd/files/2077052/lng.log.tar.gz)

**Comment by kulpreet** [comment]
Whoa. Upgrade to 4.2 totally fixes it. The same use case as above, resulted in a payment without errors

```

ubuntu@ip-172-30-4-176:~$ lncli sendpayment --pay_req=lntb1m1pd3spmjpp5fp28zz47scgjm789jfdux5nd9ukwez8n6tj2nkp6gjtxprjxakgsdqqcqzyswe5rhex7krc94tjgf84n9m374ngzrgkkusjq90006xrpv2jj3fxy5rlepsl4epwpecqq0jyj9nv2v923jzw24lfu0cxzct3sh3c9tccqyygs32
{
        "payment_error": "",
        "payment_preimage": "e206ed21ebebb1060ae09c2c297f86dcb36440e012d1c6fda4131530f73df0f7",
        "payment_route": {
                "total_time_lock": 1324482,
                "total_fees": 2,
                "total_amt": 100002,
                "hops": [
                        {
                                "chan_id": 1455803972713644033,
                                "chan_capacity": 6781941,
                                "amt_to_forward": 100001,
                                "fee": 1,
                                "expiry": 1324338,
                                "amt_to_forward_msat": 100001100,
                                "fee_msat": 1100
                        },
                        {
                                "chan_id": 1455555483086684161,
                                "chan_capacity": 300000,
                                "amt_to_forward": 100000,
                                "fee": 1,
                                "expiry": 1324194,
                                "amt_to_forward_msat": 100000000,
                                "fee_msat": 1100
                        },
                        {
                                "chan_id": 1454637390874148864,
                                "chan_capacity": 12800000,
                                "amt_to_forward": 100000,
                                "expiry": 1324194,
                                "amt_to_forward_msat": 100000000
                        }
                ],
                "total_fees_msat": 2200,
                "total_amt_msat": 100002200
        }
}
```

**Comment by Roasbeef** [comment]
👍
